include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -402,8 +402,15 @@ namespace internal
                  dynamic_cast<const MappingCartesian<dim, spacedim> *>(
                    &mapping))
         {
-          mapping_cartesian->fill_mapping_data_for_generic_points(
-            cell, unit_points, update_flags_mapping, mapping_data);
+          if constexpr (dim == spacedim)
```

Comment:
For testing purposes, I have use `if constexpr`. If the proposed approach is fine, I will replace these by C++14-conforming templates.